src/celeritas/global/alongstep/AlongStep.hh

```diff
@@ -108,17 +109,56 @@ inline CELER_FUNCTION void along_step(MH&& msc,
         }
     }
 
-    apply_eloss(track, &local.step_limit);
-    CELER_ASSERT(local.step_limit.step > 0);
-    CELER_ASSERT(local.step_limit.action);
+    if (eloss.is_applicable(track))
+    {
+        using Energy = ParticleTrackView::Energy;
+
+        bool apply_cut = (local.step_limit.action != track.boundary_action());
+        Energy deposited
+            = eloss.calc_eloss(track, local.step_limit.step, apply_cut);
+        CELER_ASSERT(deposited <= particle.energy());
+        CELER_ASSERT(apply_cut || deposited < particle.energy());
```

Comment:
I see a case that this assert fails in demo-looop (after relaxing the max_nsteps in FieldDriver).

Reply:
I don't understand what you mean: the demo-loop unit test with adjusted parameters?

Reply:
(This was in cms2018 through a standalone GPU demo.)